lib/engine/game/g_1849/depot.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+require_relative '../../depot'
+
+module Engine
+  module Game
+    module G1849
+      class Depot < Engine::Depot
+        def min_depot_train_no_e_token
```

Comment:
This should be done in `buy_train` step. Both 18Neb and 1844 has examples of filtering trains by corporation type.

Reply:
I can do it the way it's done in 1844 if that's preferred. This is the way it was done in 1848, which is where I borrowed the code from.

Reply:
This is fine